modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -78,6 +78,41 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
         raise Proxy::DHCP::NotImplemented.new("DhcpsApi::Server#get_free_ip_address is not available on Windows Server 2008 and earlier versions.")
       end
 
+      dhcp_range = dhcpsapi.list_subnet_elements(subnet_address, DhcpsApi::DHCP_SUBNET_ELEMENT_TYPE::DhcpIpRanges)
+      dhcp_start_addr = dhcp_range.map {|r| r[:element][:start_address]}.first
```

Comment:
It's possible to create a subnet without a subnet range, so both dhcp_start_addr and dhcp_end_addr can come back as nils.

Reply:
Well, this is the range on the MS DHCP, afaik we can't create a dhcp scope without a range?

Reply:
I think it's possible when api is used to create a subnet.